File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        l = 0
        r = len(numbers)-1
        while l < r:
            if numbers[l] + numbers[r] == target:
                return [l+1, r+1]
            elif numbers[l] + numbers[r] > target:
                r -= 1
            else:
                l += 1

class Solution(object):
    def twoSum(self, numbers, target):

        l = 0
        r = len(numbers) - 1

        while (l <= r):
            sum = numbers[l] + numbers[r]
            if sum > target:
                r -= 1
            elif sum < target:
                l += 1
            else:
                return [l+1, r+1]

File: 167-two-sum-ii-input-array-is-sorted/two-sum-ii-input-array-is-sorted.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& vec1, int key) {
        int st=0, end = vec1.size()-1;
        // vector<int> ans;
        while(st < end){
            if(vec1[st]+vec1[end]==key){
                // ans.push_back(st);
                // ans.push_back(end);
                return {st+1, end+1};
                break;
            }else if(vec1[st]+vec1[end]>key){
                end--;
            }else{
                st++;
            }
        }
        return {};
    }
};